experimenter/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/FormFeatureValue/fml.ts

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Diagnostic } from "@codemirror/lint";
+import {
+  createDiagnostic,
+  documentPosition,
+  parseError,
+  TestableEditorView,
+} from "src/components/PageEditBranches/FormBranches/FormFeatureValue/helpers";
+import { getConfig_nimbusConfig_applicationConfigs_channels } from "src/types/getConfig";
+import { getExperiment_experimentBySlug_featureConfigs } from "src/types/getExperiment";
+
+const line = 1;
+const column = 2;
+
+/**
+ * Parse from FmlErrors to Diagnostic
+ */
+function parseFmlErrors(
+  view: TestableEditorView,
+  fmlErrors: unknown[],
+): Diagnostic[] {
+  const diagnostics: Diagnostic[] = [];
+
+  for (const error of fmlErrors) {
+    const err = parseError.parse(error);
+    const pos = documentPosition(view.state.doc, err.line, err.column - 1);
+
+    diagnostics.push(createDiagnostic(pos, pos, err.rawMessage));
+  }
+  return diagnostics;
+}
+
+export function fmlLinter(
+  featureConfig: getExperiment_experimentBySlug_featureConfigs,
+  channel: getConfig_nimbusConfig_applicationConfigs_channels,
+  schema: Record<string, unknown>,
+) {
+  const application = featureConfig.application;
+  return async function (view: TestableEditorView) {
+    // Todo: call endpoint with app/channel/json-able text
+    const errors = [
+      parseError.parse({
+        column: column,
+        line: line,
+        rawMessage: "oh no!",
+        highlight: "error",
+      }),
+    ];
```

Comment:
Suggestion: for demo purposes only, you may want to check if `view.state.doc.…` contains the string "error" and then construct an FML error out of that position. If not, then return no errors.

Also: it may be useful to duplicate / refactor [the code to validate that the string is a valid JSON object](https://github.com/mozilla/experimenter/pull/9154/files#diff-9c4dd8943255d31a8a717fb5bf9150e1abb9b71dc4e5deae61d063615391d816R93-R134)— though happy if this is a followup issue.

Reply:
I think the validation will be better as a followup issue. I'll include the "error" checking so we can demo this :D

Reply:
yeah the JSON validation should be pulled out into a function that both of these linters will call.